lib/util.c

```diff
@@ -1818,6 +1839,23 @@ EXPORTED int hex_to_bin(const char *hex, size_t hexlen, void *bin)
     return (unsigned char *)v - (unsigned char *)bin;
 }
 
+EXPORTED int buf_hex_to_bin(struct buf *bin, const char *hex, size_t hexlen)
+{
+    if (hex == NULL)
+        return -1;
+    if (hexlen == 0)
+        hexlen = strlen(hex);
+
+    size_t newlen = bin->len + hexlen / 2;
```

Comment:
I did only add the checks that are relevant for buf_hex_to_bin to work properly until it calls hex_to_bin. But I now duplicated the check.

Reply:
I was confused because I couldn't see the changes, but I was looking at the "util: ..." commit.  Now I see you've applied these changes, but in the "imapurl: ..." commit instead.  Which is surprising, did you accidentally squash them into the wrong commit?